### ISSUE TITLE
Report error if attempting to bind from nested expression

### ISSUE THREAD

**Issue by rbrush** [rbrush/clara-rules#144]
Clara should report a meaningful error when attempting to bind results from a nested expression. For example, this error is malformed since one cannot bind to the result of an or:

```clj
(defrule malformed-example
  [?stuff <- [:or
            [max-discount :from [Discount]]
            [max-discount :from [Discount]]]]
  =>
  (println ?stuff))
```

This does not report an error when building the rule session, which can lead to unexpected behavior.

**Comment by rbrush** [comment]
So the above looks like a valid form to Clara because the expression following <- is just interpreted as a fact type. So in the above example the entire expression ```[:or [max-discount ...]]``` is viewed as a single instance and is compared to for types! 

Clara supports arbitrary values for types but it seems pretty unlikely someone would actually want to use a complex expression for one. It's probably safe to throw an error if the apparent type is Clojure vector or sequence of some sort.

**Comment by rbrush** [comment]
This is fixed in the above commit. Closing and targeting for 0.9.2
